### PR TITLE
fix: added privacy prompt for firefox

### PR DESCRIPTION
**Pull request by rebelchris**
## Changes

Firefox has very strict privacy rules and thus we need to comply and offer a clear first interaction privacy setup.
As we don't provide any way for user to turn it off/deny it we need to present a "uninstall extension" prompt.
*ONLY SHOWS ON FF

Video of usecase:
https://github.com/dailydotdev/apps/assets/554874/8f9f1469-a18e-4ed6-b0f4-348ca5ab3d55

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

AS-429 #done

### Preview domain
https://as-429-firefox-uninstaller.preview.app.daily.dev

### CODE REVIEW

**Review by capJavert — APPROVED** [review]

File: packages/shared/src/components/modals/firefoxPrivacy/FirefoxPrivacyModal.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    <Modal {...props} shouldCloseOnOverlayClick={false} size={ModalSize.Medium}>
+      <Modal.Body>
+        <h1 className="text-center font-bold typo-title3">
+          Your privacy stays yours
+        </h1>
+        <div className="mb-6 mt-4 text-center text-text-secondary typo-callout">
+          <p>
+            We collect user data to provide a personalized experience on
+            daily.dev. Instead of just collecting data right away, we would like
+            to ask for your approval. We promise to never misuse it.
```
Comment (omBratteng) [inline packages/shared/src/components/modals/firefoxPrivacy/FirefoxPrivacyModal.tsx:33]: Maybe add link to our privacy policy?

Comment (rebelchris) [reply packages/shared/src/components/modals/firefoxPrivacy/FirefoxPrivacyModal.tsx:33]: I chatted to Nimrod before he said it was fine not doing so but raised it again 🙏

Comment (omBratteng) [reply packages/shared/src/components/modals/firefoxPrivacy/FirefoxPrivacyModal.tsx:33]: I predict FF might react that there's no link to the policy.

Comment (rebelchris) [reply packages/shared/src/components/modals/firefoxPrivacy/FirefoxPrivacyModal.tsx]: Added for now :)

**Review by omBratteng — APPROVED** [review]

**Pull request merged**